post.md

```diff
@@ -0,0 +1,702 @@
+data in registers and memory changes. Where possible, actual
+values appear, where value depends on initial value, the complete
+transformation is recorded. Of course, it's only prototype and
+more advanced instructions are not yet supported, but it's
+already promising. With proper jump support, we could reduce
+multiple control switches into a minimal set of transformations
+for given goal, even separate it from the rest of code.
+However far it would go, doing a similar thing with higher
+level language seems many times more complex.
+
+For fuller and more reliable and efficient implementation
+Capstone-Keystone-Unicorn tools could be used. We may also
+consider QEMU (which allows us to dump all ongoing operations
+and changing state) and GDB integration. When something
+simpler is needed we may consider `mprotect()` Linux call
+for runtime code rebuilding and analysis.
+
+# How to start?
+
+Actually, this concept is based on very simple things. They
+may seem distant and abstract because nowadays they are
+typically covered with multiple layers of abstraction but
+machine code execution and organization are quite
+straightforward. I believe that knowing their principles can
+help in navigating through modern overly complex systems
+and getting a broader understanding of computing.
+
+Of course, x86 instruction set is quite broad and sometimes
+introduce very complex mechanisms. However for understanding
+how computations are divided into independent pieces and how
+do they communicate we need just to scratch the surface. Same
+with memory and process management — there is the complex theory
+behind them, but interface we get from the OS is relatively
+easy.
+
+We have CPU — computation unit and RAM for temporary data
+storage. We are usually separated from all other devices but
+we use OS interface, which reuses concepts we use for
+those two. Generally speaking, everything that happens in 
+a computer is series of passing data between components and
+transforming them in between.
+
+For example, when we display a web page we can think of it:
+
+1. we have given an address in memory.
+2. transform it into request conforming standard.
+3. pass a request to the networking device
+4. accept response
+5. transform text into an image
+6. pass the image to graphics card
+
+The job of application is just to transform the data and
+inform OS where is the product, what it is and where to pass
+it. RAM is a medium for that exchange as well as storage for
+middle products. RAM also store instructions how to perform
+those transformations. Assembly language is a textual
+representation of codes understood by CPU.
+
+# Memory
+
+We can think of RAM as a function. Every byte in memory has
+its ordinal number. We use it to read or change its value.
+Bytes are usually accessed in groups of 4(32bit) or 8(64-bit).
+
+X86 CPU never two such groups at a time, that's why it has
+own memory called registers. In modern x86 architecture, there
```

Comment:
@BartekLew first sentence in this paragraph sounds weird.

Reply:
:+1: